src/backend/api/Fusion.Resources.Api/Authentication/ResourcesLocalClaimsTransformation.cs

```diff
@@ -36,15 +42,38 @@ public async Task<IEnumerable<Claim>> TransformUserAsync(ClaimsPrincipal princip
             return claims;
         }
 
-        private static Task ApplyResourceOwnerForDepartmentClaimIfUserIsResourceOwnerAsync(FusionFullPersonProfile profile, List<Claim> claims)
+        private async Task ApplyResourceOwnerForDepartmentClaimIfUserIsResourceOwnerAsync(FusionFullPersonProfile profile, List<Claim> claims)
         {
-            if (profile.IsResourceOwner && !string.IsNullOrEmpty(profile.FullDepartment))
-            {
-                claims.Add(new Claim(ResourcesClaimTypes.ResourceOwnerForDepartment, profile.FullDepartment));
+            // This will now point to incorrect department. We need to use the roles on the profile, to see scoped manager responsebility.
+            // Leaving in for reference.
+            //if (profile.IsResourceOwner && !string.IsNullOrEmpty(profile.FullDepartment))
+            //{
+            //    claims.Add(new Claim(ResourcesClaimTypes.ResourceOwnerForDepartment, profile.FullDepartment));
+            //}
+
+            if (profile.Roles is null) {
+                throw new InvalidOperationException("Roles must be loaded on the profile for the claims transformer to work.");
             }
 
-            return Task.CompletedTask;
-        }
+            var managerRoles = profile.Roles
+                .Where(x => string.Equals(x.Name, "Fusion.LineOrg.Manager", StringComparison.OrdinalIgnoreCase))
+                .Where(x => !string.IsNullOrEmpty(x.Scope?.Value))
+                .Select(x => x.Scope?.Value!)
+                .ToList();
+
+            // Got a list of sap id's, need to resolve them to the full department to keep consistent.
+            logger.LogInformation($"Found user responsible for [{managerRoles.Count}] org units [{string.Join(",", managerRoles)}]");
+
+            foreach (var orgUnitId in managerRoles)
+            {
+                var orgUnit = await mediator.Send(new ResolveLineOrgUnit(orgUnitId));
+                if (orgUnit?.FullDepartment != null)
+                {
+                    claims.Add(new Claim(ResourcesClaimTypes.ResourceOwnerForDepartment, orgUnit.FullDepartment));
+                    logger.LogInformation($"Adding claim for {orgUnitId} -> [{orgUnit.FullDepartment}]");
```

Comment:
Excessive logging? Unsure if this is a problem or not just a thought. I assume this runs on every request

Reply:
The intention for this is to help troubleshoot potential early issues. So will leave in for now, then we can remove as things stabalize

Reply:
Good idea